### PR TITLE
Core/Packets: implement GAMEOBJECT_TYPE_UI_LINK

### LINKED ISSUE #20913: Hero's Call Board in Ironforge does not offer any quests
**Description:** Hero's Call Board in Ironforge does not offer any quests

**Expected behaviour:** board should offer breadcrumb quests according to character's level.

**Steps to reproduce the problem:**

1. Create a level 30 Alliance character
2. Travel to Ironforge
3. Go to the Hero's Call Board in front of the AH
4. Notice no exclamation marks on top of it
5. Click the board
6. Nothing happens

**Branch(es):** master

**TC rev. hash/commit:** 7334419

**TDB version:** 720.00_2017_04_18

**Operating system:** MacOS

screen shot 2017-11-17 at 1 53 10 am

### PR DESCRIPTION
**Pull request by DoctorKraft**
**Changes proposed:**

-  Implement GAMEOBJECT_TYPE_UI_LINK for gob with type 48
Get UI for Obliterium Forge or Hero's Call Board with Adventure Journal

**Target branch(es):**

- [X] master

**Issues addressed:** Closes #20913 

**Tests performed:** Build & tested in-game

**Known issues and TODO list:**

- [ ] Radius not implemented (if anyone have an idea)

### CODE REVIEW

File: src/server/game/Server/Packets/GameObjectPackets.h [hunk 1]
```diff
@@ -120,6 +120,17 @@ namespace WorldPackets
...
+
+        class GameObjectUIAction final : public ServerPacket
+        {
+        public:
+            GameObjectUIAction() : ServerPacket(SMSG_GAME_OBJECT_UI_ACTION, 16 + 4) { }
+
+            WorldPacket const* Write() override;
+
+            ObjectGuid ObjectGUID;
+            uint32 UILink = 0;
```
Comment (Shauren) [inline src/server/game/Server/Packets/GameObjectPackets.h]: This one should be int32

File: src/server/game/Server/Packets/GameObjectPackets.cpp [hunk 2]
```diff
@@ -69,3 +69,11 @@ WorldPacket const* WorldPackets::GameObject::GameObjectCustomAnim::Write()
 
     return &_worldPacket;
 }
+
+WorldPacket const* WorldPackets::GameObject::GameObjectUIAction::Write()
+{
+    _worldPacket << ObjectGUID;
+    _worldPacket << UILink;
```
Comment (Shauren) [inline src/server/game/Server/Packets/GameObjectPackets.cpp]: always write the type for ints/floats (`_worldPacket << int32(UILink);`)

**Review by Shauren — CHANGES_REQUESTED** [review]
And radius - isnt that simply the interaction distance limit?

**Comment by DoctorKraft** [comment]
@Shauren yeah i think same, but i don't have any idea where to implement this. It's in GridNotifier?

**Comment by joschiwald** [comment]
take a look at GameObject::GetInteractionDistance

**Comment by DoctorKraft** [comment]
I have tried one but not working. For example i set 50 yards, when icon for interact are grayed you can do everything it's always same range (only when icon are not grayed). The structure contains also a spellfocus...

**Comment by joschiwald** [comment]
it looks like that all gameobjects have the default interaction range

**Comment by joschiwald** [comment]
merging in the next days

**Comment by Killyana** [comment]
Gobs interaction range will be fixed by https://github.com/TrinityCore/TrinityCore/pull/18917

**Pull request merged**